DancingGoat/Helpers/Extensions/HtmlHelperExtensions.cs

```diff
@@ -46,17 +48,19 @@ public static MvcHtmlString AssetImage(this HtmlHelper htmlHelper, Asset asset,
         /// <summary>
         /// Displays a <see cref="DateTime"/> in a formatted manner.
         /// </summary>
-        /// <param name="htmlHelper">HTML helper.</param>
+        /// <param name="htmlHelper">HTML helper</param>
+        /// <param name="expression">The expression of the model property</param>
         /// <param name="dateTime">The <see cref="DateTime"/> to format</param>
         /// <param name="format">The formatting character</param>
-        public static MvcHtmlString DateTimeFormatted(this HtmlHelper htmlHelper, DateTime? dateTime, string format)
+        /// <remarks>The TValue generic parameter is chosen instead of DateTime just to save views from falling to exceptions. With TValue, the views will get rendered, only this helper method will return an empty <see cref="MvcHtmlString"/>.</remarks>
+        public static MvcHtmlString DateTimeFormattedFor<TModel, TValue>(this HtmlHelper<TModel> htmlHelper, Expression<Func<TModel, TValue>> expression, string format)
         {
-            if (!dateTime.HasValue || string.IsNullOrEmpty(format))
+            if (expression.ReturnType != typeof(DateTime?) && expression.ReturnType != typeof(DateTime))
```

Comment:
This 'mechanism' is meant to handle the different types of input (other than DateTime and DateTime?). The 'handling' consists of rendering nothing, which I find most appropriate for handling problems in such presentation layer. We simply have no ways of enforcing the DateTime in the signature while having such 'safe' error handling at the same time. Maybe I'm wrong, but I've spent quite some time with researching this issue earlier.